include/ensmallen_bits/nsga2/nsga2_impl.hpp

```diff
@@ -396,9 +398,7 @@ inline bool NSGA2::Dominates(
 inline void NSGA2::CrowdingDistanceAssignment(const std::vector<size_t>& front,
                                               std::vector<double>& crowdingDistance)
 {
-  if (front.size() > 0)
-  {
-    for (size_t elem: front)
```

Comment:
Continuing on the "big fish", this causes this redundant if clause because of that empty vector.

Reply:
https://github.com/haris989/NSGA-II/blob/7680fd70c5202d0673aa26b9fb50a8e9c387623b/NSGA%20II.py#L76, notice there are no checks